src/legacy/ui/public/agg_types/buckets/terms.js

```diff
@@ -248,61 +270,99 @@ export const termsBucketAgg = new BucketAggType({
         {
           display: i18n.translate('common.ui.aggTypes.buckets.terms.orderAscendingTitle', {
             defaultMessage: 'Ascending',
           }),
-          val: 'asc'
-        }
+          val: 'asc',
+        },
       ],
-      write: _.noop // prevent default write, it's handled by orderAgg
+      write: _.noop, // prevent default write, it's handled by orderAgg
     },
     {
       name: 'orderBy',
-      write: _.noop // prevent default write, it's handled by orderAgg
+      write: _.noop, // prevent default write, it's handled by orderAgg
     },
     {
       name: 'otherBucket',
       default: false,
-      editor: otherBucketTemplate,
-      write: _.noop
-    }, {
+      editorComponent: SwitchParamEditor,
+      dataTestSubj: 'otherBucketSwitch',
+      displayLabel: i18n.translate('common.ui.aggTypes.otherBucket.groupValuesLabel', {
+        defaultMessage: 'Group other values in separate bucket',
+      }),
+      displayToolTip: i18n.translate('common.ui.aggTypes.otherBucket.groupValuesTooltip', {
+        defaultMessage:
+          'Values not in the top N are grouped in this bucket. To include documents with missing values, enable \'Show missing values\'.',
+      }),
+      write: _.noop,
+    },
+    {
       name: 'otherBucketLabel',
+      type: 'string',
       default: i18n.translate('common.ui.aggTypes.buckets.terms.otherBucketLabel', {
         defaultMessage: 'Other',
       }),
-      write: _.noop
-    }, {
+      displayName: i18n.translate('common.ui.aggTypes.otherBucket.labelForOtherBucketLabel', {
+        defaultMessage: 'Label for other bucket',
+      }),
+      shouldShow: agg => agg.params.otherBucket,
+      write: _.noop,
+    },
+    {
       name: 'missingBucket',
       default: false,
-      write: _.noop
-    }, {
+      editorComponent: SwitchParamEditor,
+      dataTestSubj: 'missingBucketSwitch',
+      displayLabel: i18n.translate('common.ui.aggTypes.otherBucket.showMissingValuesLabel', {
+        defaultMessage: 'Show missing values',
+      }),
+      displayToolTip: i18n.translate('common.ui.aggTypes.otherBucket.showMissingValuesTooltip', {
+        defaultMessage:
+          'Only works for fields of type "string". When enabled, include documents with missing ' +
+          'values in the search. If this bucket is in the top N, it appears in the chart. ' +
+          'If not in the top N, and you enable "Group other values in separate bucket", ' +
+          'Elasticsearch adds the missing values to the "other" bucket.',
+      }),
+      disabled: agg => !isType('string')(agg),
+      write: _.noop,
+    },
+    {
       name: 'missingBucketLabel',
       default: i18n.translate('common.ui.aggTypes.buckets.terms.missingBucketLabel', {
         defaultMessage: 'Missing',
         description: `Default label used inside of charts for documents missing a specific field.
           Can be seen when creating a chart with a terms aggregation and select the "Show missing values"
```

Comment:
This sentence is a little confusing.  Can it be:

Default label used in charts when documents are missing a field.  Visible when you create a chart with a terms aggregation and enable "Show missing values".

Reply:
@gchaps done 👌